src/unread/UnreadCards.js

```diff
@@ -30,11 +33,37 @@ class UnreadCards extends PureComponent<Props> {
   };
 
   handleTopicPress = (stream: string, topic: string) => {
-    setTimeout(() => this.props.dispatch(doNarrow(topicNarrow(stream, topic))));
+    const { bulkSelection, handleTopicSelect } = this.props;
+    if (bulkSelection === null) {
+      setTimeout(() => this.props.dispatch(doNarrow(topicNarrow(stream, topic))));
+    } else {
+      handleTopicSelect(stream, topic);
+    }
+  };
+
+  handleTopicLongPress = (stream: string, topic: string) => {
+    const { handleTopicSelect } = this.props;
+    handleTopicSelect(stream, topic);
+  };
+
+  getBulkSelectionStatus = (stream: string, topic: string): boolean => {
+    const { bulkSelection } = this.props;
+    if (bulkSelection === null) {
+      return false;
+    }
+    const narrow = JSON.stringify(topicNarrow(stream, topic));
+    return bulkSelection.includes(narrow);
   };
 
   render() {
-    const { conversations, unreadStreamsAndTopics, ...restProps } = this.props;
+    const {
+      conversations,
+      unreadStreamsAndTopics,
+      bulkSelection,
+      // eslint-disable-next-line no-unused-vars
+      handleTopicSelect,
```

Comment:
This comment is present here because although `handleTopicSelect` is unused, I don't want it going into `restProps`.

Reply:
Makes sense. Let's put that in the code comment, though. 😉